backend/serializers.py

```diff
@@ -69,6 +70,84 @@ class Meta:
                   'gender')
 
 
+class RecruitmentResultAggregateSerializer(serializers.ModelSerializer[Any]):
+    candidates_count = serializers.SerializerMethodField()
+    first_cycle_threshold = serializers.SerializerMethodField()
+    second_cycle_threshold = serializers.SerializerMethodField()
+    third_cycle_threshold = serializers.SerializerMethodField()
+
+    def get_recruitments_filters(self, obj: Any) -> Any:
+        pass
+
+    def get_candidates_count(self, obj: Any) -> int:
+        recruitments = Recruitment.objects.\
+            filter(**self.get_recruitments_filters(obj))
+        return RecruitmentResult.objects\
+            .filter(recruitment__in=recruitments).count()
+
+    def get_cycle_threshold(self, obj: Any, cycle: int) -> Any:
+        recruitments_filters = self.get_recruitments_filters(obj)
+        recruitments_filters['round'] = cycle
+        recruitments = Recruitment.objects.filter(**recruitments_filters)
+        recruitment_results = RecruitmentResult.objects.filter(
+            recruitment__in=recruitments, result='Signed'
+        )
+        if recruitment_results:
+            result = recruitment_results.aggregate(Min('points')).\
+                get('points__min')
+            print(result)
+            return result
+        return None
+
+    def get_first_cycle_threshold(self, obj: Any) -> Any:
+        return self.get_cycle_threshold(obj, 1)
+
+    def get_second_cycle_threshold(self, obj: Any) -> Any:
+        return self.get_cycle_threshold(obj, 2)
+
+    def get_third_cycle_threshold(self, obj: Any) -> Any:
```

Comment:
Jednak chyba nie możemy zakładać na sztywno ilości cykli i przydało by się, żeby zwracany był na przykład obiekt
{ nr_cyklu: próg}
czyli byłoby to:
{
1: 920,
2: 900,
}

Reply:
A to nie jest tak, że na całym AGH sa maksymalnie trzy cykle? W przypadku mniejszej ilosci byłby zwracany null

Reply:
hm a nie jest właśnie tak, że może być więcej?